lua/neotest-kotlin/treesitter/kotest-query.lua

```diff
@@ -82,6 +82,24 @@ return [[
 ;; -- todo BEHAVIOR SPEC --
 ;; -- todo FREE SPEC --
 ;; -- todo WORD SPEC --
+
+; Matches "context" `when` { /** body **/ }
+; Matches "context" When { /** body **/ }
+
+(infix_expression
+  (string_literal) @namespace.name
+  (simple_identifier) @function_name (#any-of? @function_name "`when`" "When")
+  (lambda_literal)
+) @namespace.definition
```

Comment:
TIL: `#any-of` allows you to match more than one string! Makes it easy here when it supports `when` and When for namespaces